src/frontend/src/binder/expr/mod.rs

```diff
@@ -565,13 +565,8 @@ pub fn bind_data_type(data_type: &AstDataType) -> Result<DataType> {
         AstDataType::Timestamp(true) => DataType::Timestamptz,
         AstDataType::Interval => DataType::Interval,
         AstDataType::Array(datatype) => DataType::List(Box::new(bind_data_type(datatype)?)),
-        AstDataType::Char(..) => {
-            return Err(ErrorCode::NotImplemented(
-                "CHAR is not supported, please use VARCHAR instead\n".to_string(),
-                None.into(),
-            )
-            .into())
-        }
+        // workaround to bind char type to varchar.
+        AstDataType::Char(..) => DataType::Varchar,
```

Comment:
`char(n)` has different semantic from (unlimited) `varchar`. People creating `char(n)` columns would get wrong results when we process them as `varchar`.

Reply:
Indeed, let's leave it unchanged and try to resolve it in the upstream side.